### PR TITLE
Test that locks are properly released in case of mode switch.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Check that locks are cleared on a mode switch together with active transactions.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@lutovich can you please take a look?

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Test( timeout = TIMEOUT_MILLIS )
+    public void lockCleanupOnModeSwitch() throws Throwable
+    {
+        Label testLabel = Label.label( "testLabel" );
+        HighlyAvailableGraphDatabase master = cluster.getMaster();
+        createNodeOnMaster( testLabel, master );
+
+        HighlyAvailableGraphDatabase slave = cluster.getAnySlave();
+        ClusterManager.RepairKit repairKit = takeExclusiveLockAndKillSlave( testLabel, slave );
```
Comment (lutovich) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java:61]: Shouldn't here be smth like `cluster.await(masterSeesSlavesAsAvailable(1))` to make sure that shutdown is noticed by other members?

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        // repair of slave and new mode switch cycle on all members
+        repairKit.repair();
+
+        HighlyAvailableGraphDatabase clusterMaster = cluster.getMaster();
```
Comment (lutovich) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java]: Similarly to the prev comment, maybe add `cluster.await(masterSeesAllSlavesAsAvailable())`?

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        Node node = getNode( db, testLabel );
+        transaction.acquireWriteLock( node );
+        return cluster.shutdown( db );
+    }
+
+    private void createNodeOnMaster( Label testLabel, HighlyAvailableGraphDatabase master )
+    {
+        try ( Transaction transaction = master.beginTx() )
+        {
+            Node ignored = master.createNode( testLabel );
```
Comment (lutovich) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java]: `Node ignored =` can be omitted

**Comment by lutovich** [comment]
@MishaDemianenko review done, couple comments

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java (hunk 1 above)
Comment (MishaDemianenko) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java:61]: We already have that kind of check in https://github.com/neo4j/neo4j/blob/3.0/enterprise/ha/src/test/java/org/neo4j/kernel/impl/ha/ClusterManager.java#L1184

**Comment by MishaDemianenko** [comment]
@lutovich updated

**Pull request merged**